libs/ktem/ktem/reasoning/prompt_optimization/fewshot_rewrite_question.py

```diff
@@ -0,0 +1,115 @@
+import langchain_core.messages
+from ktem.llms.manager import llms
+from ktem.reasoning.prompt_optimization.rewrite_question import (
+    DEFAULT_REWRITE_PROMPT,
+    RewriteQuestionPipeline,
+)
+from langchain.prompts import (
+    ChatPromptTemplate,
+    FewShotChatMessagePromptTemplate,
+    SemanticSimilarityExampleSelector,
+)
+from langchain_community.vectorstores import Chroma
+from langchain_openai import OpenAIEmbeddings
+from theflow.settings import settings as flowsettings
+
+from kotaemon.base import AIMessage, Document, HumanMessage, Node, SystemMessage
+from kotaemon.llms import ChatLLM
+
+
+class FewshotRewriteQuestionPipeline(RewriteQuestionPipeline):
+    """Rewrite user question
+
+    Args:
+        llm: the language model to rewrite question
+        rewrite_template: the prompt template for llm to paraphrase a text input
+        lang: the language of the answer. Currently support English and Japanese
+    """
+
+    llm: ChatLLM = Node(default_callback=lambda _: llms.get_default())
+    rewrite_template: str = DEFAULT_REWRITE_PROMPT
+    lang: str = "English"
+    final_prompt: ChatPromptTemplate
+
+    def create_example_selector(
+        self, examples, k: int = getattr(flowsettings, "N_PROMPT_OPT_EXAMPLES", 0)
+    ):
+        if not examples:
+            return None
```

Comment:
Seems like `examples` is only used for this. If you mean "if there is no example, don't use a selector" then this logic should be handled outside of `create_example_selector`.